### DISCUSSION TITLE
Issue: VLS Not Found Error (101)

### DISCUSSION THREAD

**Discussion by Ranne314** [vlang/v#22747] [Installation Issues]
I am having trouble installing VLS on a Windows x86_64 Laptop. This is a fresh install of Vlang:
Here is my Shell:

PS C:\Users\Derek> v ls
    v ls error: Language server is not installed nor found. (101)

    - If you are using this for the first time, please run
        `v ls --install` first to download and install VLS.
    - If you are using a custom version of VLS, check if
        the specified path exists and is a valid executable.
    - If you have an existing installation of VLS, be sure
        to remove "vls.config.json" and "bin" located inside
        "%USERPROFILE%\.vls" and re-install.

    If none of the options listed have solved your issue,
        please report it at https://github.com/vlang/v/issues

    C:/Users/Derek/AppData/Local/Temp/v_0/vls.01JBRNWQZEPKZYSHXHPWBW80GK.tmp.c:14373: at print_backtrace: Backtrace
    C:/Users/Derek/AppData/Local/Temp/v_0/vls.01JBRNWQZEPKZYSHXHPWBW80GK.tmp.c:48535: by main__VlsUpdater_cli_error
    C:/Users/Derek/AppData/Local/Temp/v_0/vls.01JBRNWQZEPKZYSHXHPWBW80GK.tmp.c:48646: by main__main
    C:/Users/Derek/AppData/Local/Temp/v_0/vls.01JBRNWQZEPKZYSHXHPWBW80GK.tmp.c:49185: by wmain
    008df968 : by ???
    008dfacb : by ???
    7ffc414a259d : by ???

PS C:\Users\Derek> v ls --install
    > Finding prebuilt executables from GitHub release..
    > Executable found for this system. Downloading...
    > Verifying checksum...
    > Updating permissions...
    > Updating vls.config.json...
    > VLS was updated to version: 0.0.1.dd8c004

PS C:\Users\Derek> v ls
    v ls error: Language server is not installed nor found. (101)

    - If you are using this for the first time, please run
        `v ls --install` first to download and install VLS.
    - If you are using a custom version of VLS, check if
        the specified path exists and is a valid executable.
    - If you have an existing installation of VLS, be sure
        to remove "vls.config.json" and "bin" located inside
        "%USERPROFILE%\.vls" and re-install.

    If none of the options listed have solved your issue,
        please report it at https://github.com/vlang/v/issues

    C:/Users/Derek/AppData/Local/Temp/v_0/vls.01JBRNWQZEPKZYSHXHPWBW80GK.tmp.c:14373: at print_backtrace: Backtrace
    C:/Users/Derek/AppData/Local/Temp/v_0/vls.01JBRNWQZEPKZYSHXHPWBW80GK.tmp.c:48535: by main__VlsUpdater_cli_error
    C:/Users/Derek/AppData/Local/Temp/v_0/vls.01JBRNWQZEPKZYSHXHPWBW80GK.tmp.c:48646: by main__main
    C:/Users/Derek/AppData/Local/Temp/v_0/vls.01JBRNWQZEPKZYSHXHPWBW80GK.tmp.c:49185: by wmain
    008df968 : by ???
    008dfacb : by ???
    7ffc414a259d : by ???

PS C:\Users\Derek>

**Comment by Ranne314** [comment]
plz answer 🙏 🙏 🙏

**Comment by Ranne314** [comment]
You are Sigma if you answer 🗣🗣🗣🗣🗣🔥🔥🔥🔥🔥🔥🔥🔥🔥

**Reply by JalonSolov** [reply]
Which version of V do you have?

**Reply by Ranne314** [reply]
$ v --version
V 0.4.8 6e9a66d

**Reply by JalonSolov** [reply]
I took a look at the source for the command, and it turns out it doesn't matter - the source needs to be updated for this to work correctly.  It is still pointing to the old, deprecated vls, when it should be pointing to the v-analyzer.

Please go to https://github.com/vlang/v-analyzer and follow the installation instructions there.

**Reply by Ranne314** [reply]
ok

**Reply by Ranne314** [reply]
Will it work with: https://marketplace.visualstudio.com/items?itemName=vlanguage.vscode-vlang

**Reply by JalonSolov** [reply]
Yes, and that plugin should properly install v-analyzer for you - you don't have to install it yourself.

**Reply by Ranne314** [reply]
It didn`t on Windows x86_64

**Comment by Wajinn** [comment]
This really should be fixed, because this can clearly cause confusion for new users. Particularly as to which is the official V one. Many people use Visual Studio Code or VSCodium, so better to make it easy on people. Both extensions are still up.

Seems like both extensions should point to one sever. If people get confused about using VLS or V-analyzer, if they were both pointing to the same server, then it wouldn't make a difference.  There is also something to be said about using the name V-analyzer, because various people would expect it to be VLS (standard type name). Only way that I can see to resolve that, is for both names to point to the same server. Then at some long away future point, get rid of one of the names.

**Reply by popisdead** [reply]
Guys the LSP should be #1 priority. V-analyzer keep on crash both on vs-code and on nvim.
It also creates multiple instances of it, making impossible to code with it for more than 5 minutes.
